src/Campaigns/Blocks/CampaignDonations/render.php

```diff
@@ -37,9 +37,14 @@
     ->joinDonationMeta(DonationMetaKeys::AMOUNT, 'amountMeta')
     ->joinDonationMeta(DonationMetaKeys::FIRST_NAME, 'donorName')
     ->leftJoin('give_donors', 'donorIdMeta.meta_value', 'donors.id', 'donors')
-    ->orderBy($sortBy === 'top-donations' ? 'amountMeta.meta_value' : 'donation.ID', 'DESC')
     ->limit($attributes['donationsPerPage'] ?? 5);
 
+if ($sortBy === 'top-donations') {
+    $query->orderByRaw('CAST(amountMeta.meta_value AS DECIMAL) DESC');
+} else {
+    $query->orderBy('donation.ID', 'DESC');
+}
```

Comment:
Seems all good here, just a proposal of keeping it inline as:
```php
$query->orderByRaw($sortBy === 'top-donations' ? 'CAST(amountMeta.meta_value AS DECIMAL) DESC' : 'donation.ID DESC');
```

Reply:
Resolved in 0996ede3c6228391d57213a8c03cf2fc7ff78f72